packages/trader/src/AppV2/Containers/Trade/trade-types.tsx

```diff
@@ -32,35 +33,34 @@ const TradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTradeTyp
     const [is_editing, setIsEditing] = React.useState<boolean>(false);
     const trade_types_ref = React.useRef<HTMLDivElement>(null);
 
-    const { onMount, onUnmount } = useTraderStore();
-
-    const createArrayFromCategories = (data: any): TItem[] => {
+    const createArrayFromCategories = (data: TTradeTypesProps['trade_types']): TItem[] => {
         const result: TItem[] = [];
 
-        data.forEach((category: { value: string; text: string }) => {
+        data.forEach(category => {
             result.push({
                 id: category.value,
-                title: category.text,
+                title: category.text ?? '',
             });
         });
 
         return result;
     };
 
-    const saved_other_trade_types = JSON.parse(localStorage.getItem('other_trade_types') ?? '[]');
-    const saved_pinned_trade_types = JSON.parse(localStorage.getItem('pinned_trade_types') ?? '[]');
+    const saved_pinned_trade_types_string: string = localStorage.getItem('pinned_trade_types') ?? '[]';
```

Comment:
We can also use `useLocalStorageData` hook. It returns data itself and methods of changing it in localStorage 
I used it while was working on user onboarding guide, was really convenient

Reply:
It doesn't work well with useLocalStorageData as it goes on effect -> setstate loop.